query-service-impl/src/main/java/org/hypertrace/core/query/service/prometheus/PrometheusBasedRequestHandler.java

```diff
@@ -36,13 +36,16 @@
   private Optional<String> startTimeAttributeName;
   private PrometheusViewDefinition prometheusViewDefinition;
 
-  PrometheusBasedRequestHandler(String name, Config requestHandlerConfig) {
+  PrometheusBasedRequestHandler(
+      String name,
+      Config requestHandlerConfig,
+      PrometheusRestClientFactory prometheusRestClientFactory) {
     this.name = name;
     this.processConfig(requestHandlerConfig);
     this.queryRequestEligibilityValidator =
         new QueryRequestEligibilityValidator(prometheusViewDefinition);
     this.requestToPromqlConverter = new QueryRequestToPromqlConverter(prometheusViewDefinition);
-    this.prometheusRestClient = PrometheusRestClientFactory.get().getPrometheusClient(name);
+    this.prometheusRestClient = prometheusRestClientFactory.getPrometheusClient(name);
```

Comment:
Was responding to the other one before reviewing this - see https://github.com/hypertrace/query-service/pull/119#discussion_r768704268 (the two part registration/lookup isn't needed)